tests/js/server/aql/aql-index-choice.js

```diff
@@ -0,0 +1,513 @@
+        assertEqual([], indexNode.projections);
+        assertFalse(indexNode.indexCoversProjections);
+        assertEqual(1, indexNode.indexes.length);
+        let index = indexNode.indexes[0];
+        assertEqual("persistent", index.type);
+        assertEqual(["uid"], index.fields);
+      });
+    },
+    
+    testSingleIndexSingleFilterAndSort2: function() {
+      db[cn].ensureIndex({ type: "persistent", fields: ["uid"] });
+      
+      [
+        `FOR doc IN ${cn} FILTER doc.uid == 1234 SORT doc.uid RETURN doc`,
+        `FOR doc IN ${cn} FILTER doc.uid > 1234 SORT doc.uid RETURN doc`,
+        `FOR doc IN ${cn} FILTER doc.uid >= 1234 SORT doc.uid RETURN doc`
+      ].forEach((q) => {
+        let nodes = AQL_EXPLAIN(q).plan.nodes;
+        assertEqual(1, nodes.filter((n) => n.type === 'IndexNode').length);
+        assertEqual(0, nodes.filter((n) => n.type === 'SortNode').length);
+        let indexNode = nodes.filter((n) => n.type === 'IndexNode')[0];
+        assertEqual([], indexNode.projections);
+        assertFalse(indexNode.indexCoversProjections);
+        assertEqual(1, indexNode.indexes.length);
+        let index = indexNode.indexes[0];
+        assertEqual("persistent", index.type);
+        assertEqual(["uid"], index.fields);
+      });
+    },
+    
+    testMultipleIndexesSingleFilterAndSort: function() {
+      db[cn].ensureIndex({ type: "persistent", fields: ["uid"] });
+      db[cn].ensureIndex({ type: "persistent", fields: ["uid", "dt"] });
+      
+      [
+        `FOR doc IN ${cn} FILTER doc.uid == 1234 SORT doc.dt RETURN doc`,
+      ].forEach((q) => {
+        let nodes = AQL_EXPLAIN(q).plan.nodes;
+        assertEqual(1, nodes.filter((n) => n.type === 'IndexNode').length);
+        assertEqual(0, nodes.filter((n) => n.type === 'SortNode').length);
+        let indexNode = nodes.filter((n) => n.type === 'IndexNode')[0];
+        assertEqual([], indexNode.projections);
+        assertFalse(indexNode.indexCoversProjections);
+        assertEqual(1, indexNode.indexes.length);
+        let index = indexNode.indexes[0];
+        assertEqual("persistent", index.type);
+        assertEqual(["uid", "dt"], index.fields);
+      });
+
+      [
+        `FOR doc IN ${cn} FILTER doc.uid > 1234 SORT doc.dt RETURN doc`,
+        `FOR doc IN ${cn} FILTER doc.uid >= 1234 SORT doc.dt RETURN doc`
+      ].forEach((q) => {
+        let nodes = AQL_EXPLAIN(q).plan.nodes;
+        assertEqual(1, nodes.filter((n) => n.type === 'IndexNode').length);
+        assertEqual(1, nodes.filter((n) => n.type === 'SortNode').length);
+        let indexNode = nodes.filter((n) => n.type === 'IndexNode')[0];
+        assertEqual([], indexNode.projections);
+        assertFalse(indexNode.indexCoversProjections);
+        assertEqual(1, indexNode.indexes.length);
+        let index = indexNode.indexes[0];
+        assertEqual("persistent", index.type);
+        assertEqual(["uid", "dt"], index.fields);
+      });
+    },
```

Comment:
How about another test case where we filter on `dt` and sort on `uid`? That would be covered by the index, so we should not have a SortNode.

Reply:
Added